hermetic/src/cli/tester.rs

```diff
@@ -236,6 +310,54 @@ async fn wait_for_job(
     })
     .await
 }
+
+// waits for the simulate manager pod to fail or succeed
+async fn wait_for_simulation(
+    client: Client,
+    namespace: &str,
+    timeout_seconds: u32,
+) -> Result<(JobResult, String)> {
+    loop_with_max_timeout(timeout_seconds, MAX_KUBE_WATCH_TIMEOUT, move |timeout| {
+        let client = client.clone();
+        Box::pin(async move {
+            let pods: Api<Pod> = Api::namespaced(client.clone(), namespace);
+            let wp = WatchParams::default()
+                .labels("job-name=simulate-manager")
+                .timeout(timeout);
+            let mut stream = pods.watch(&wp, "0").await?.boxed();
+            while let Some(status) = stream.try_next().await? {
+                debug!("pod watch event {:#?}", status);
+                match status {
+                    WatchEvent::Added(pod) | WatchEvent::Modified(pod) => {
+                        trace!("found pod: {}", pod.name_unchecked());
+                        debug!("pod status: {:#?}", pod.status);
+
+                        if let Some(status) = &pod.status {
+                            if let Some(phase) = &status.phase {
+                                if phase == "Succeeded" {
```

Comment:
Well what the heck... the only comment that was at all interesting (or pedantic?) disappeared... so submitting it again.

Nothing to change, but as an FYI you can do some fancy stuff with `match`. I find myself considering it when I have nested `if let` statements. For example,

```rust

match pod.status.and_then(|s| s.phase).as_deref() {
    Some("Suceeded") => {
        // woo
    }
    Some("Failed") => {
        //boo
    }
    Some(_) => {
        unreachable!();
    }
    None => {}
};
```

You can also use [conditionals](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#extra-conditionals-with-match-guards) like `Some(v) if v == "Succeeded"` but I personally don't use those a ton.